The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9748953264/job/26905099655) [(plain)](https://github.com/rust-lang/rust/commit/45b0f68e34c37e5e5f47a1474ad87ebdb099af4b/checks/26905099655/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
## Running ui tests in tests/pass for x86_64-pc-windows-gnu
   Compiler: "MIRI_ENV_VAR_TEST"="0" "MIRI_TEMP"="/tmp/miri-uitest-All1Ya" "RUST_BACKTRACE"="1" /checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/miri "--error-format=json" "--sysroot=/checkout/obj/build/x86_64-unknown-linux-gnu/miri-sysroot" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "x86_64-pc-windows-gnu" "--out-dir" OUT_DIR

FAILED TEST: tests/pass/panic/std-panic-locations.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp/miri-uitest-All1Ya" RUST_BACKTRACE="1" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/miri" "--error-format=json" "--sysroot=/checkout/obj/build/x86_64-unknown-linux-gnu/miri-sysroot" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "x86_64-pc-windows-gnu" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/ui/tests/pass/panic" "tests/pass/panic/std-panic-locations.rs" "--edition" "2021"
error: pass test got exit status: 1, but expected 0

error: actual output differed from expected
Execute `./miri test --bless` to update `tests/pass/panic/std-panic-locations.stderr` to the actual output
Execute `./miri test --bless` to update `tests/pass/panic/std-panic-locations.stderr` to the actual output
--- tests/pass/panic/std-panic-locations.stderr
+++ <stderr output>
+error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
+  --> RUSTLIB/panic_unwind/src/gcc.rs:LL:CC
+   |
+LL |     return uw::_Unwind_RaiseException(exception_param) as u32;
+   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't call foreign function `_Unwind_RaiseException` on OS `windows`
+   = help: if this is a basic API commonly used on this target, please report an issue with Miri
+   = help: if this is a basic API commonly used on this target, please report an issue with Miri
+   = help: however, note that Miri does not aim to support every FFI function out there; for instance, we will not support APIs for things such as GUIs, scripting languages, or databases
+   = note: inside `panic_unwind::imp::panic` at RUSTLIB/panic_unwind/src/gcc.rs:LL:CC
+   = note: inside `panic_unwind::__rust_start_panic` at RUSTLIB/panic_unwind/src/lib.rs:LL:CC
+   = note: inside `std::panicking::rust_panic` at RUSTLIB/std/src/panicking.rs:LL:CC
+   = note: inside `std::panicking::rust_panic_with_hook` at RUSTLIB/std/src/panicking.rs:LL:CC
+   = note: inside `std::panicking::rust_panic_with_hook` at RUSTLIB/std/src/panicking.rs:LL:CC
+   = note: inside closure at RUSTLIB/std/src/panicking.rs:LL:CC
+   = note: inside `std::sys::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at RUSTLIB/std/src/sys/backtrace.rs:LL:CC
+note: inside closure
+  --> $DIR/std-panic-locations.rs:LL:CC
+   |
+LL |     assert_panicked(|| nope.unwrap());
+LL |     assert_panicked(|| nope.unwrap());
+   |                        ^^^^^^^^^^^^^
+   = note: inside `std::panicking::r#try::do_call::<{closure@$DIR/std-panic-locations.rs:LL:CC}, ()>` at RUSTLIB/std/src/panicking.rs:LL:CC
+   = note: inside `std::panicking::r#try::<(), {closure@$DIR/std-panic-locations.rs:LL:CC}>` at RUSTLIB/std/src/panicking.rs:LL:CC
+   = note: inside `std::panic::catch_unwind::<{closure@$DIR/std-panic-locations.rs:LL:CC}, ()>` at RUSTLIB/std/src/panic.rs:LL:CC
+note: inside `main::assert_panicked::<{closure@$DIR/std-panic-locations.rs:LL:CC}>`
+   |
+LL |         std::panic::catch_unwind(f).unwrap_err();
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+note: inside `main`
---
+


error: there were 1 unmatched diagnostics that occurred outside the testfile and had no pattern
    Error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
full stderr:
full stderr:
error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
##[error]  --> /checkout/library/panic_unwind/src/gcc.rs:72:12
   |
LL |     return uw::_Unwind_RaiseException(exception_param) as u32;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't call foreign function `_Unwind_RaiseException` on OS `windows`
   = help: if this is a basic API commonly used on this target, please report an issue with Miri
   = help: if this is a basic API commonly used on this target, please report an issue with Miri
   = help: however, note that Miri does not aim to support every FFI function out there; for instance, we will not support APIs for things such as GUIs, scripting languages, or databases
   = note: inside `panic_unwind::imp::panic` at /checkout/library/panic_unwind/src/gcc.rs:72:12: 72:55
   = note: inside `panic_unwind::__rust_start_panic` at /checkout/library/panic_unwind/src/lib.rs:100:5: 100:24
   = note: inside `std::panicking::rust_panic` at /checkout/library/std/src/panicking.rs:857:25: 857:48
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:821:5: 821:24
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:821:5: 821:24
   = note: inside closure at /checkout/library/std/src/panicking.rs:663:13: 668:14
   = note: inside `std::sys::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys/backtrace.rs:171:18: 171:21
note: inside closure
  --> tests/pass/panic/std-panic-locations.rs:24:24
   |
LL |     assert_panicked(|| nope.unwrap());
LL |     assert_panicked(|| nope.unwrap());
   |                        ^^^^^^^^^^^^^
   = note: inside `std::panicking::r#try::do_call::<{closure@tests/pass/panic/std-panic-locations.rs:24:21: 24:23}, ()>` at /checkout/library/std/src/panicking.rs:553:40: 553:43
   = note: inside `std::panicking::r#try::<(), {closure@tests/pass/panic/std-panic-locations.rs:24:21: 24:23}>` at /checkout/library/std/src/panicking.rs:517:19: 517:88
   = note: inside `std::panic::catch_unwind::<{closure@tests/pass/panic/std-panic-locations.rs:24:21: 24:23}, ()>` at /checkout/library/std/src/panic.rs:350:14: 350:33
note: inside `main::assert_panicked::<{closure@tests/pass/panic/std-panic-locations.rs:24:21: 24:23}>`
   |
LL |         std::panic::catch_unwind(f).unwrap_err();
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `main`
---



FAILED TEST: tests/pass/panic/nested_panic_caught.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp/miri-uitest-All1Ya" RUST_BACKTRACE="1" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/miri" "--error-format=json" "--sysroot=/checkout/obj/build/x86_64-unknown-linux-gnu/miri-sysroot" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "x86_64-pc-windows-gnu" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/ui/tests/pass/panic" "tests/pass/panic/nested_panic_caught.rs" "--edition" "2021"
error: pass test got exit status: 1, but expected 0

error: actual output differed from expected
Execute `./miri test --bless` to update `tests/pass/panic/nested_panic_caught.stderr` to the actual output
Execute `./miri test --bless` to update `tests/pass/panic/nested_panic_caught.stderr` to the actual output
--- tests/pass/panic/nested_panic_caught.stderr
+++ <stderr output>
 thread 'main' panicked at $DIR/nested_panic_caught.rs:LL:CC:
 once
 note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
 note: in Miri, you may have to set `-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
-thread 'main' panicked at $DIR/nested_panic_caught.rs:LL:CC:
+error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
+  --> RUSTLIB/panic_unwind/src/gcc.rs:LL:CC
-stack backtrace:
+   |



error: there were 1 unmatched diagnostics that occurred outside the testfile and had no pattern
    Error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
full stderr:
thread 'main' panicked at tests/pass/panic/nested_panic_caught.rs:20:5:
once
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: in Miri, you may have to set `-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
##[error]  --> /checkout/library/panic_unwind/src/gcc.rs:72:12
   |
LL |     return uw::_Unwind_RaiseException(exception_param) as u32;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't call foreign function `_Unwind_RaiseException` on OS `windows`
   = help: if this is a basic API commonly used on this target, please report an issue with Miri
   = help: if this is a basic API commonly used on this target, please report an issue with Miri
   = help: however, note that Miri does not aim to support every FFI function out there; for instance, we will not support APIs for things such as GUIs, scripting languages, or databases
   = note: inside `panic_unwind::imp::panic` at /checkout/library/panic_unwind/src/gcc.rs:72:12: 72:55
   = note: inside `panic_unwind::__rust_start_panic` at /checkout/library/panic_unwind/src/lib.rs:100:5: 100:24
   = note: inside `std::panicking::rust_panic` at /checkout/library/std/src/panicking.rs:857:25: 857:48
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:821:5: 821:24
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:821:5: 821:24
   = note: inside closure at /checkout/library/std/src/panicking.rs:663:13: 668:14
   = note: inside `std::sys::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys/backtrace.rs:171:18: 171:21
note: inside `double`
  --> tests/pass/panic/nested_panic_caught.rs:20:5
   |
LL |     panic!("once");
LL |     panic!("once");
   |     ^^^^^^^^^^^^^^
note: inside closure
  --> tests/pass/panic/nested_panic_caught.rs:24:29
   |
LL |     assert!(catch_unwind(|| double()).is_err());
   |                             ^^^^^^^^
   = note: inside `std::panicking::r#try::do_call::<{closure@tests/pass/panic/nested_panic_caught.rs:24:26: 24:28}, ()>` at /checkout/library/std/src/panicking.rs:553:40: 553:43
   = note: inside `std::panicking::r#try::<(), {closure@tests/pass/panic/nested_panic_caught.rs:24:26: 24:28}>` at /checkout/library/std/src/panicking.rs:517:19: 517:88
   = note: inside `std::panic::catch_unwind::<{closure@tests/pass/panic/nested_panic_caught.rs:24:26: 24:28}, ()>` at /checkout/library/std/src/panic.rs:350:14: 350:33
  --> tests/pass/panic/nested_panic_caught.rs:24:13
   |
   |
LL |     assert!(catch_unwind(|| double()).is_err());
   = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

---



FAILED TEST: tests/pass/panic/catch_panic.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp/miri-uitest-All1Ya" RUST_BACKTRACE="1" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/miri" "--error-format=json" "--sysroot=/checkout/obj/build/x86_64-unknown-linux-gnu/miri-sysroot" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "x86_64-pc-windows-gnu" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/ui/tests/pass/panic" "tests/pass/panic/catch_panic.rs" "--edition" "2021"
error: pass test got exit status: 1, but expected 0

error: actual output differed from expected
Execute `./miri test --bless` to update `tests/pass/panic/catch_panic.stderr` to the actual output
Execute `./miri test --bless` to update `tests/pass/panic/catch_panic.stderr` to the actual output
--- tests/pass/panic/catch_panic.stderr
+++ <stderr output>
 thread 'main' panicked at $DIR/catch_panic.rs:LL:CC:
 Hello from std::panic
 note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
 note: in Miri, you may have to set `-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
-Caught panic message (&str): Hello from std::panic
+error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
-thread 'main' panicked at $DIR/catch_panic.rs:LL:CC:
+  --> RUSTLIB/panic_unwind/src/gcc.rs:LL:CC
+   |
-Caught panic message (String): Hello from std::panic: 1
-Caught panic message (String): Hello from std::panic: 1
+LL |     return uw::_Unwind_RaiseException(exception_param) as u32;
-thread 'main' panicked at $DIR/catch_panic.rs:LL:CC:
+   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't call foreign function `_Unwind_RaiseException` on OS `windows`
-Hello from std::panic_any: 2
-Caught panic message (String): Hello from std::panic_any: 2
+   = help: if this is a basic API commonly used on this target, please report an issue with Miri
-thread 'main' panicked at $DIR/catch_panic.rs:LL:CC:
-thread 'main' panicked at $DIR/catch_panic.rs:LL:CC:
+   = help: however, note that Miri does not aim to support every FFI function out there; for instance, we will not support APIs for things such as GUIs, scripting languages, or databases
-Box<dyn Any>
-Failed to get caught panic message.
+   = note: inside `panic_unwind::imp::panic` at RUSTLIB/panic_unwind/src/gcc.rs:LL:CC
-thread 'main' panicked at $DIR/catch_panic.rs:LL:CC:
+   = note: inside `panic_unwind::__rust_start_panic` at RUSTLIB/panic_unwind/src/lib.rs:LL:CC
+   = note: inside `panic_unwind::__rust_start_panic` at RUSTLIB/panic_unwind/src/lib.rs:LL:CC
-Hello from core::panic
+   = note: inside `std::panicking::rust_panic` at RUSTLIB/std/src/panicking.rs:LL:CC
-Caught panic message (&str): Hello from core::panic
+   = note: inside `std::panicking::rust_panic_with_hook` at RUSTLIB/std/src/panicking.rs:LL:CC
-thread 'main' panicked at $DIR/catch_panic.rs:LL:CC:
+   = note: inside closure at RUSTLIB/std/src/panicking.rs:LL:CC
-Hello from core::panic: 5
+   = note: inside `std::sys::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at RUSTLIB/std/src/sys/backtrace.rs:LL:CC
-Caught panic message (String): Hello from core::panic: 5
-thread 'main' panicked at $DIR/catch_panic.rs:LL:CC:
+note: inside closure
-index out of bounds: the len is 3 but the index is 4
+  --> $DIR/catch_panic.rs:LL:CC
+  --> $DIR/catch_panic.rs:LL:CC
-Caught panic message (String): index out of bounds: the len is 3 but the index is 4
+   |
-thread 'main' panicked at $DIR/catch_panic.rs:LL:CC:
+LL |     test(None, |_old_val| std::panic!("Hello from std::panic"));
-attempt to divide by zero
+   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Caught panic message (&str): attempt to divide by zero
+note: inside `do_panic_counter::<{closure@$DIR/catch_panic.rs:LL:CC}>`
-thread 'main' panicked at RUSTLIB/core/src/ptr/const_ptr.rs:LL:CC:
-align_offset: align is not a power-of-two
+   |
-Caught panic message (&str): align_offset: align is not a power-of-two
+LL |     do_panic(old_val);
---
+LL |         do_panic_counter(do_panic)
-Caught panic message (&str): assertion failed: false
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
-Success!
+   = note: inside `<std::panic::AssertUnwindSafe<{closure@$DIR/catch_panic.rs:LL:CC}> as std::ops::FnOnce<()>>::call_once` at RUSTLIB/core/src/panic/unwind_safe.rs:LL:CC

error: there were 1 unmatched diagnostics that occurred outside the testfile and had no pattern
error: there were 1 unmatched diagnostics that occurred outside the testfile and had no pattern
    Error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
full stderr:
thread 'main' panicked at tests/pass/panic/catch_panic.rs:50:27:
Hello from std::panic
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: in Miri, you may have to set `-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
##[error]  --> /checkout/library/panic_unwind/src/gcc.rs:72:12
   |
LL |     return uw::_Unwind_RaiseException(exception_param) as u32;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't call foreign function `_Unwind_RaiseException` on OS `windows`
   = help: if this is a basic API commonly used on this target, please report an issue with Miri
   = help: if this is a basic API commonly used on this target, please report an issue with Miri
   = help: however, note that Miri does not aim to support every FFI function out there; for instance, we will not support APIs for things such as GUIs, scripting languages, or databases
   = note: inside `panic_unwind::imp::panic` at /checkout/library/panic_unwind/src/gcc.rs:72:12: 72:55
   = note: inside `panic_unwind::__rust_start_panic` at /checkout/library/panic_unwind/src/lib.rs:100:5: 100:24
   = note: inside `std::panicking::rust_panic` at /checkout/library/std/src/panicking.rs:857:25: 857:48
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:821:5: 821:24
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:821:5: 821:24
   = note: inside closure at /checkout/library/std/src/panicking.rs:663:13: 668:14
   = note: inside `std::sys::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys/backtrace.rs:171:18: 171:21
note: inside closure
  --> tests/pass/panic/catch_panic.rs:50:27
   |
LL |     test(None, |_old_val| std::panic!("Hello from std::panic"));
LL |     test(None, |_old_val| std::panic!("Hello from std::panic"));
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: inside `do_panic_counter::<{closure@tests/pass/panic/catch_panic.rs:50:16: 50:26}>`
   |
LL |     do_panic(old_val);
   |     ^^^^^^^^^^^^^^^^^
note: inside closure
note: inside closure
  --> tests/pass/panic/catch_panic.rs:98:9
   |
LL |         do_panic_counter(do_panic)
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: inside `<std::panic::AssertUnwindSafe<{closure@tests/pass/panic/catch_panic.rs:96:45: 96:47}> as std::ops::FnOnce<()>>::call_once` at /checkout/library/core/src/panic/unwind_safe.rs:272:9: 272:19
   = note: inside `std::panicking::r#try::do_call::<std::panic::AssertUnwindSafe<{closure@tests/pass/panic/catch_panic.rs:96:45: 96:47}>, ()>` at /checkout/library/std/src/panicking.rs:553:40: 553:43
   = note: inside `std::panicking::r#try::<(), std::panic::AssertUnwindSafe<{closure@tests/pass/panic/catch_panic.rs:96:45: 96:47}>>` at /checkout/library/std/src/panicking.rs:517:19: 517:88
   = note: inside `std::panic::catch_unwind::<std::panic::AssertUnwindSafe<{closure@tests/pass/panic/catch_panic.rs:96:45: 96:47}>, ()>` at /checkout/library/std/src/panic.rs:350:14: 350:33
note: inside `test::<{closure@tests/pass/panic/catch_panic.rs:50:16: 50:26}>`
   |
LL |       let res = catch_unwind(AssertUnwindSafe(|| {
   |  _______________^
LL | |         let _string = "LEAKED FROM CLOSURE".to_string();
---



FAILED TEST: tests/pass/panic/thread_panic.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp/miri-uitest-All1Ya" RUST_BACKTRACE="1" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/miri" "--error-format=json" "--sysroot=/checkout/obj/build/x86_64-unknown-linux-gnu/miri-sysroot" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "x86_64-pc-windows-gnu" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/ui/tests/pass/panic" "tests/pass/panic/thread_panic.rs" "--edition" "2021"
error: pass test got exit status: 1, but expected 0

error: actual output differed from expected
Execute `./miri test --bless` to update `tests/pass/panic/thread_panic.stderr` to the actual output
Execute `./miri test --bless` to update `tests/pass/panic/thread_panic.stderr` to the actual output
--- tests/pass/panic/thread_panic.stderr
+++ <stderr output>
 thread '<unnamed>' panicked at $DIR/thread_panic.rs:LL:CC:
 Hello!
 note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
 note: in Miri, you may have to set `-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
-thread 'childthread' panicked at $DIR/thread_panic.rs:LL:CC:
+error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
+  --> RUSTLIB/panic_unwind/src/gcc.rs:LL:CC


error: there were 1 unmatched diagnostics that occurred outside the testfile and had no pattern
error: there were 1 unmatched diagnostics that occurred outside the testfile and had no pattern
    Error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
full stderr:
thread '<unnamed>' panicked at tests/pass/panic/thread_panic.rs:6:35:
Hello!
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: in Miri, you may have to set `-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
##[error]  --> /checkout/library/panic_unwind/src/gcc.rs:72:12
   |
LL |     return uw::_Unwind_RaiseException(exception_param) as u32;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't call foreign function `_Unwind_RaiseException` on OS `windows`
   = help: if this is a basic API commonly used on this target, please report an issue with Miri
   = help: if this is a basic API commonly used on this target, please report an issue with Miri
   = help: however, note that Miri does not aim to support every FFI function out there; for instance, we will not support APIs for things such as GUIs, scripting languages, or databases
   = note: BACKTRACE on thread `unnamed-1`:
   = note: inside `panic_unwind::imp::panic` at /checkout/library/panic_unwind/src/gcc.rs:72:12: 72:55
   = note: inside `panic_unwind::__rust_start_panic` at /checkout/library/panic_unwind/src/lib.rs:100:5: 100:24
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:821:5: 821:24
   = note: inside closure at /checkout/library/std/src/panicking.rs:663:13: 668:14
   = note: inside closure at /checkout/library/std/src/panicking.rs:663:13: 668:14
   = note: inside `std::sys::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys/backtrace.rs:171:18: 171:21
note: inside closure
  --> tests/pass/panic/thread_panic.rs:6:35
   |
   |
LL |     let result = thread::spawn(|| panic!("Hello!")).join().unwrap_err();
   = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

---



FAILED TEST: tests/pass/panic/concurrent-panic.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp/miri-uitest-All1Ya" RUST_BACKTRACE="1" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1/bin/miri" "--error-format=json" "--sysroot=/checkout/obj/build/x86_64-unknown-linux-gnu/miri-sysroot" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "x86_64-pc-windows-gnu" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/ui/tests/pass/panic" "tests/pass/panic/concurrent-panic.rs" "-Zmiri-preemption-rate=0" "--edition" "2021"
error: pass test got exit status: 1, but expected 0

error: actual output differed from expected
Execute `./miri test --bless` to update `tests/pass/panic/concurrent-panic.stderr` to the actual output
Execute `./miri test --bless` to update `tests/pass/panic/concurrent-panic.stderr` to the actual output
--- tests/pass/panic/concurrent-panic.stderr
+++ <stderr output>
 Thread 1 starting, will block on mutex
 Thread 1 reported it has started
... 2 lines skipped ...
 note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
 note: in Miri, you may have to set `-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
-Thread 2 blocking on thread 1
+error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
+  --> RUSTLIB/panic_unwind/src/gcc.rs:LL:CC
-Unlocking mutex
+   |
-thread '<unnamed>' panicked at $DIR/concurrent-panic.rs:LL:CC:
-thread '<unnamed>' panicked at $DIR/concurrent-panic.rs:LL:CC:
+LL |     return uw::_Unwind_RaiseException(exception_param) as u32;
-panic in thread 1
+   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't call foreign function `_Unwind_RaiseException` on OS `windows`
+   |
-Thread 2 has exited
+   = help: if this is a basic API commonly used on this target, please report an issue with Miri



error: there were 1 unmatched diagnostics that occurred outside the testfile and had no pattern
    Error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
full stderr:
Thread 1 starting, will block on mutex
Thread 1 reported it has started
thread '<unnamed>' panicked at tests/pass/panic/concurrent-panic.rs:66:13:
thread '<unnamed>' panicked at tests/pass/panic/concurrent-panic.rs:66:13:
panic in thread 2
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: in Miri, you may have to set `-Zmiri-env-forward=RUST_BACKTRACE` for the environment variable to have an effect
error: unsupported operation: can't call foreign function `_Unwind_RaiseException` on OS `windows`
##[error]  --> /checkout/library/panic_unwind/src/gcc.rs:72:12
   |
LL |     return uw::_Unwind_RaiseException(exception_param) as u32;
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't call foreign function `_Unwind_RaiseException` on OS `windows`
   = help: if this is a basic API commonly used on this target, please report an issue with Miri
   = help: if this is a basic API commonly used on this target, please report an issue with Miri
   = help: however, note that Miri does not aim to support every FFI function out there; for instance, we will not support APIs for things such as GUIs, scripting languages, or databases
   = note: BACKTRACE on thread `unnamed-2`:
   = note: inside `panic_unwind::imp::panic` at /checkout/library/panic_unwind/src/gcc.rs:72:12: 72:55
   = note: inside `panic_unwind::__rust_start_panic` at /checkout/library/panic_unwind/src/lib.rs:100:5: 100:24
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:821:5: 821:24
   = note: inside closure at /checkout/library/std/src/panicking.rs:663:13: 668:14
   = note: inside closure at /checkout/library/std/src/panicking.rs:663:13: 668:14
   = note: inside `std::sys::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys/backtrace.rs:171:18: 171:21
note: inside closure
  --> tests/pass/panic/concurrent-panic.rs:66:13
   |
LL |             panic!("panic in thread 2");
---
Error: 
   0: ui tests in tests/pass for x86_64-pc-windows-gnu failed
   1: tests failed

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
error: test failed, to rerun pass `--test ui`
Caused by:
Caused by:
  process didn't exit successfully: `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools/x86_64-unknown-linux-gnu/release/deps/ui-5c8e20648e59180a pass --quiet` (exit status: 1)
  local time: Mon Jul  1 18:30:18 UTC 2024
  network time: Mon, 01 Jul 2024 18:30:18 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/tools/miri/src/shims/windows/foreign_items.rs
+++ b/src/tools/miri/src/shims/windows/foreign_items.rs
@@ -758,6 +758,22 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.write_null(dest)?;
             }
 
+            "_Unwind_RaiseException" => {
+                // This is not formally part of POSIX, but it is very wide-spread on POSIX systems.
+                // It was originally specified as part of the Itanium C++ ABI:
+                // https://itanium-cxx-abi.github.io/cxx-abi/abi-eh.html#base-throw.
+                // MinGW implements _Unwind_RaiseException on top of SEH exceptions.
+                if this.tcx.sess.target.env != "gnu" {
+                    throw_unsup_format!(
+                        "`_Unwind_RaiseException` is not supported on non-MinGW Windows",
+                    );
+                }
+                // This function looks and behaves excatly like miri_start_unwind.
+                let [payload] = this.check_shim(abi, Abi::C { unwind: true }, link_name, args)?;
+                this.handle_miri_start_unwind(payload)?;
+                return Ok(EmulateItemResult::NeedsUnwind);
+            }
+
             _ => return Ok(EmulateItemResult::NotSupported),
         }
 

